app/upgrades/v07/upgrades.go

```diff
@@ -0,0 +1,34 @@
+package v07
+
+import (
+	wasmkeeper "github.com/CosmWasm/wasmd/x/wasm/keeper"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+
+	poekeeper "github.com/confio/tgrade/x/poe/keeper"
+	poetypes "github.com/confio/tgrade/x/poe/types"
+	twasmkeeper "github.com/confio/tgrade/x/twasm/keeper"
+)
+
+func CreateUpgradeHandler(
+	wasmKeeper twasmkeeper.Keeper,
+	poeKeeper poekeeper.ContractSource,
+) upgradetypes.UpgradeHandler {
+	return func(ctx sdk.Context, plan upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		ctx.Logger().Info("Migrating to set tfi factory migrator address")
+		govVotingContractAddr, err := poeKeeper.GetPoEContractAddress(ctx, poetypes.PoEContractTypeValidatorVoting)
+		if err != nil {
+			return nil, sdkerrors.Wrap(err, "validator voting contract address")
+		}
+		tfiFactoryAddr, err := sdk.AccAddressFromBech32(plan.Info) // using the info field to transport the address
+		if err != nil {
+			return nil, sdkerrors.Wrap(err, "tfi contract address")
+		}
+
+		govPermission := wasmkeeper.NewGovPermissionKeeper(&wasmKeeper.Keeper)
+		err = govPermission.UpdateContractAdmin(ctx, tfiFactoryAddr, govVotingContractAddr, govVotingContractAddr)
```

Comment:
👍🏼

Reply:
Wouldn't make sense to take the opportunity and update the other two contracts (tfi-pair and trusted-token) migrate admins as well?

It will require some changes, though, as three addresses would need to be passed / set in the plan.

Reply:
Another thing we could do in passing, is update the tfi-factory contract to the patched version already.

Shouldn't be necessary, though.

Reply:
Good thinking. I did not do this as https://github.com/confio/tfi/issues/71 is not done. There would be no end to this as new pairs can be created any time

Reply:
> There would be no end to this as new pairs can be created any time

Good point. Luckily we don't need to patch those.